src/prefect/cli/cloud/__init__.py

```diff
@@ -318,9 +318,24 @@ async def login(
 
     profiles = load_profiles()
     current_profile = get_settings_context().profile
+    env_var_api_key = PREFECT_API_KEY.value()
 
-    if key and PREFECT_API_KEY.value() == key:
```

Comment:
Just because we have a `PREFECT_API_KEY` env var set and it's equal to the `key` passed in to `prefect cloud login -k pnu_xyz`, and we are authenticated does not mean that we are using that particular profile, which I think is the prime of the `login` command.